modules/javafx.graphics/src/main/java/com/sun/javafx/css/TransitionDefinition.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.javafx.css;
+
+import javafx.animation.Interpolator;
+import javafx.css.StyleableProperty;
+import javafx.util.Duration;
+import java.util.Objects;
+
+/**
+ * {@code TransitionDefinition} describes how a {@link StyleableProperty} changes from one value to
+ * another when its value is changed implicitly by the CSS subsystem. The transition can be smooth,
+ * for example using linear or Bézier interpolation, or discrete using stepwise interpolation.
+ *
+ * @param propertyName the CSS property name, or "all" to target any property
+ * @param duration duration of the transition
+ * @param delay delay after which the transition is started; if negative, the transition starts
+ *              immediately, but will appear to have begun at an earlier point in time
+ * @param interpolator interpolator for the transition
+ */
+public record TransitionDefinition(String propertyName, Duration duration,
+                                   Duration delay, Interpolator interpolator) {
```

Comment:
I've tried using `long` nanosecond fields. The amount of conversions doesn't change, it merely shifts around the location where the conversions happen. Since there's no obvious benefit, I prefer to use `javafx.util.Duration` for clarity.

Additionally, I've introduced two helper methods in `TransitionTimer` (`nanosToMillis` and `millisToNanos`) to make sure that conversion is done correctly in all places.